atomic_reactor/plugins/input_osv3.py

```diff
@@ -36,55 +37,71 @@ def run(self):
+            self.plugins_json.setdefault('{0}_plugins'.format(typ), [])
+
+        # XXX: Remove the two if blocks after we stop using super old osbs:(
+        pull_plugin = self.get_plugin('prebuild_plugins', PullBaseImagePlugin.key)
+        if not pull_plugin:
             self.log.warning("%s is missing in prebuild_plugins - please update your osbs-client!",
                              PullBaseImagePlugin.key)
             pull_plugin = { "name": PullBaseImagePlugin.key }
-            plugins_json['prebuild_plugins'].insert(0, pull_plugin)
+            self.plugins_json['prebuild_plugins'].insert(0, pull_plugin)
 
-        try:
-            change_plugin = [p for p in plugins_json['prebuild_plugins']
-                             if p.get('name', None) == 'change_source_registry'][0]
-        except IndexError:
-            pass
-        else:
+        change_plugin = self.get_plugin('prebuild_plugins', 'change_source_registry')
+        if change_plugin:
             if 'registry_uri' in change_plugin.get('args', {}):
                 pull_plugin.setdefault('args', {})['parent_registry'] = \
                     change_plugin['args']['registry_uri']
             if 'insecure_registry' in change_plugin.get('args', {}):
                 pull_plugin.setdefault('args', {})['parent_registry_insecure'] = \
                     change_plugin['args']['insecure_registry']
-            plugins_json['prebuild_plugins'].remove(change_plugin)
+            self.plugins_json['prebuild_plugins'].remove(change_plugin)
 
         if 'parent_registry' not in pull_plugin.get('args', {}):
             self.log.error("source registry is not configured")
 
-        input_json = {
-            'source': {
-                'provider': 'git',
-                'uri': git_url,
-                'provider_params': {'git_commit': git_ref}
-            },
-            'image': image,
-            'target_registries': [target_registry] if target_registry else None,
-            'target_registries_insecure': True,  # FIXME: create plugin for this
-            'openshift_build_selflink': build_json.get('metadata', {}).get('selfLink', None)
-        }
-        input_json.update(plugins_json)
-
-        self.log.debug("build json: %s", input_json)
-
-        return input_json
+        push_plugin = self.get_plugin('postbuild_plugins', TagAndPushPlugin.key)
+        if not push_plugin and self.target_registry:
+            self.log.warning("%s is missing in postbuild_plugins - please update your osbs-client!",
+                             TagAndPushPlugin.key)
+            push_plugin = { "name": TagAndPushPlugin.key }
+            self.plugins_json['postbuild_plugins'].insert(0, push_plugin)
+
+        if push_plugin and self.target_registry:
+            push_plugin.setdefault('args', {}).setdefault('registries', {})
+            if not push_plugin['args']['registries']:
+                push_plugin['args']['registries'][self.target_registry] = {"insecure": True}
```

Comment:
The reasoning here was that we'll move away from using the target registry (defined in the openshift env variable) eventually, can't really see the use case for using both configuration sources with different registries.